Updates for SymfonyCon talk

willdurand opened the pull request:
Hi!

Here is a Work In Progress for the talk  Lukas and I will give at SymfonyCon.

William

willdurand committed:
Remove FSCHateoasBundle

willdurand committed:
Add BazingaHateoasBundle

willdurand committed:
Update deps

willdurand committed:
Configure Hateoas relations for Note/NoteCollection resources

willdurand committed:
Switch to stable NelmioApiDocBundle version

willdurand committed:
Use NelmioApiDocBundle FormExtension

+ rearrange URI sections

lsmith77 commented on README.md:
```diff
@@ -198,8 +198,6 @@ It comes pre-configured with the following bundles:
 
   * [**FOSRestBundle**][16] - Adds rest functionality
 
-  * [**FSCHateoasBundle**][18] - Adds HATEOAS support
```
why remove instead of replace with BazingaHateoasBundle ?

willdurand replied:
Forgot to update the doc while adding the BazingaHateoasBundle..

lsmith77 replied:
can you fix that .. i will then merge

willdurand replied:
done.

it is up to you, but you could keep this PR open until I push everything.

lsmith77 replied:
ok .. lets keep it open ..

willdurand committed:
Update README

willdurand committed:
Fix CS

willdurand committed:
Introduce Event class

Configure Hateoas 'embed' relations on Notes, show exclusion strategy,
dynmamic rel

willdurand commented:
#### http://localhost:8000/notes/0.json

``` json
{
    "message": "Note 0",
    "_links": {
        "self": {
            "href": "http://localhost:8000/notes/0"
        }
    }
}
```

#### http://localhost:8000/notes/1.json

``` json
{
    "message": "Note 1",
    "_links": {
        "self": {
            "href": "http://localhost:8000/notes/1"
        }
    },
    "_embedded": {
        "associated_events": [
            {
                "name": "SymfonyCon",
                "date": "2013-12-12T00:00:00+0100"
            },
            {
                "name": "Christmas Day",
                "date": "2013-12-25T00:00:00+0100"
            }
        ]
    }
}
```

willdurand committed:
Add HautelookTemplatedUriBundle

willdurand commented:
#### http://localhost:8000/notes.json

``` json
{
    "notes": [
        {
            "secret": "6r3Fe3c893oDXVGVkWhQ8RbIjZcyn410/MF3+skHRHgzIcYalQx00h+aqf4LBeTH6ASqNBHsW275fWv4YG1u0g==",
            "message": "Note 0",
            "_links": {
                "self": {
                    "href": "http://localhost:8000/notes/0"
                }
            }
        },
        {
            "secret": "fmRCVeaSWpSEXUNdfaCUrzvm02MvpGaHHNJv5O5ekcGkMEL470hANDBCWWqHVpn44TIq0EBzv548tLW6mOv8vA==",
            "message": "Note 1",
            "_links": {
                "self": {
                    "href": "http://localhost:8000/notes/1"
                }
            },
            "_embedded": {
                "associated_events": [
                    {
                        "name": "SymfonyCon",
                        "date": "2013-12-12T00:00:00+0100"
                    },
                    {
                        "name": "Christmas Day",
                        "date": "2013-12-25T00:00:00+0100"
                    }
                ]
            }
        }
    ],
    "limit": 5,
    "_links": {
        "self": {
            "href": "http://localhost:8000/notes"
        },
        "note": {
            "href": "http://localhost:8000/notes/{id}",
            "templated": true
        }
    }
}
```

willdurand commented:
@lsmith77 WDYT?

lsmith77 commented:
looks good .. no need to get fancier.

willdurand commented:
Hopefully, Hautelook guys will have time to look at my PRs:
- https://github.com/hautelook/TemplatedUriBundle/pull/10
- https://github.com/hautelook/TemplatedUriRouter/pull/3

willdurand commented:
I will create another PR for Hautelook stuff, and maybe try to add BazingaRestExtraBundle, not sure how yet.

lsmith77 commented:
cool

adrienbrault commented on src/Acme/DemoBundle/Model/Note.php:
```diff
@@ -24,7 +24,21 @@ class Note
-}
\ No newline at end of file
+
+    public function getAssociatedEventsRel()
```
Is that just to highlight the feature ?

willdurand replied:
Yep
